plugins/GSdx/Renderers/DX11/GSDevice11.cpp

```diff
@@ -1229,26 +1259,20 @@ void GSDevice11::PSSetShader(ID3D11PixelShader* ps, ID3D11Buffer* ps_cb)
 		m_ctx->PSSetShader(ps, NULL, 0);
 	}
 
-	if(m_srv_changed)
-	{
-		m_ctx->PSSetShaderResources(0, countof(m_state.ps_srv), m_state.ps_srv);
-
-		m_srv_changed = false;
-	}
-
-	if(m_ss_changed)
-	{
-		m_ctx->PSSetSamplers(0, countof(m_state.ps_ss), m_state.ps_ss);
-
-		m_ss_changed = false;
-	}
-
 	if(m_state.ps_cb != ps_cb)
 	{
 		m_state.ps_cb = ps_cb;
 
 		m_ctx->PSSetConstantBuffers(0, 1, &ps_cb);
 	}
+
+	PSUpdateShaderState();
+}
+
+void GSDevice11::PSUpdateShaderState()
```

Comment:
I couldn't think of a good name for this function.
I hope this is acceptable.

Reply:
It looks fine to me.